src/pages/ProfilesExplorerView/components/SceneExploreServiceFlameGraph/SceneExploreServiceFlameGraph.tsx

```diff
@@ -29,6 +34,56 @@ export class SceneExploreServiceFlameGraph extends SceneObjectBase<SceneExploreS
       }),
       body: new SceneFlameGraph(),
     });
+
+    this.addActivationHandler(this.onActivate.bind(this, item));
+  }
+
+  onActivate(item?: GridItemData) {
+    if (item) {
+      this.initVariables(item);
+    }
+
+    const profileMetricVariable = findSceneObjectByClass(this, ProfileMetricVariable) as ProfileMetricVariable;
+
+    profileMetricVariable.setState({ query: ProfileMetricVariable.QUERY_SERVICE_NAME_DEPENDENT });
```

Comment:
The Scene object is also responsible for its own specific setup/teardown (like making the profile metric variable aware of the selected service)